src/kixi/stats/distribution.cljc

```diff
@@ -686,40 +701,48 @@
   Probabilities should be >= 0 and sum to 1"
   [category-probs]
   (let [[ks ps] (apply map vector category-probs)]
+    (assert (every? #(<= 0.0 % 1.0) ps) "All the probbilities must be between 0.0 and 1.0.")
     (->Categorical ks ps)))
 
 (defn multinomial
   "Returns a multinomial distribution.
   Params: {:n ∈ ℕ > 0, :probs [ℝ >= 0, ...]}
   Probabilities should be >= 0 and sum to 1"
   [{:keys [n probs]}]
+  (assert (pos-int? n) (str "n (" n ") must be a positive integer."))
+  (assert (every? #(<= 0.0 % 1.0) probs)
+          "All the probbilities must be between 0.0 and 1.0.")
   (->Multinomial n probs))
 
 (defn dirichlet
   "Returns a Dirichlet distribution.
   Params: {:alphas [ℝ >= 0, ...]}"
   [{:keys [alphas]}]
+  (assert (every? #(not (neg? %)) alphas) "All the alphas must be non-negative.")
```

Comment:
`(every? (complement neg?) alphas)` would be neater here, or `(not (some neg? alphas))`

Reply:
Okay.